### PR TITLE
Add README.md for dp, mf and optimization

### PR DESCRIPTION
**Pull request by DavdGao**
as the title says

### CODE REVIEW

File: scripts/README.md [hunk 1]
```diff
@@ -2,10 +2,43 @@
 We provide some scripts for reproducing existing algorithms with FederatedScope, which are constantly being updated.
 We greatly appreciate any [contribution](https://federatedscope.io/docs/contributor/) to FederatedScope!
 
+- [Federated Optimization Algorithms](#fed-optimization)
```
Comment (xieyxclack) [inline scripts/README.md:5]: an invalid link here: `#fed-optimization` -> `#federated-optimization-algorithm`

File: scripts/README.md [hunk 2]
```diff
@@ -2,10 +2,43 @@
...
 - [Distribute Mode](#distribute-mode)
 - [Asynchronous Training Strategy](#asynchronous-training-strategy)
 - [Graph Federated Learning](#graph-federated-learning)
 - [Attacks in Federated Learning](#attacks-in-FL)
+- [Differential Privacy in Federated Learning](#dp-in-FL)
```
Comment (xieyxclack) [inline scripts/README.md:10]: Invalid link

File: scripts/README.md [hunk 3]
```diff
@@ -2,10 +2,43 @@
...
+- [Matrix Factorization in Federated Learning](#mf-in-FL)
```
Comment (xieyxclack) [inline scripts/README.md:11]: Invalid link

File: scripts/README.md [hunk 4]
```diff
@@ -2,10 +2,43 @@
...
+```bash
+# on femnist
+bash optimization_exp_scripts/fedprox_exp_scripts/run_fedprox_femnist.sh
+# on lr
+bash optimization_exp_scripts/fedprox_exp_scripts/run_fedprox_lr.sh
+# on shakespeare
+bash optimization_exp_scripts/fedprox_exp_scripts/run_fedprox_shakespeare.sh
+```
+
+[1] Asad M, Moustafa A, Ito T. "FedOpt: Towards communication efficiency and privacy preservation in federated learning". Applied Sciences, 2020, 10(8): 2864.
```
Comment (xieyxclack) [inline scripts/README.md:39]: Move all the references to the end of this docs?

**Review by xieyxclack — COMMENTED** [review]
LGTM, and please resolve the conflicts and make sure the provided scripts is correct and executable. Thanks!

File: scripts/README.md (hunk 1 above)
Comment (DavdGao) [reply scripts/README.md]: modified accordingly

File: scripts/README.md (hunk 2 above)
Comment (DavdGao) [reply scripts/README.md]: modified accordingly

File: scripts/README.md (hunk 4 above)
Comment (DavdGao) [reply scripts/README.md]: modified accordingly

**Review by xieyxclack — APPROVED** [review]

**Pull request merged**